client/rpc_sender.go

```diff
@@ -109,6 +111,9 @@ func (s *rpcSender) Send(_ context.Context, call proto.Call) {
 		break
 	}
 	if err != nil {
-		call.Reply.Header().SetGoError(err)
+		return nil, proto.NewError(err)
 	}
+	pErr := br.Error
```

Comment:
at this point, we're on the wire, so there's really no way around this so long as our protos have an error field.